docs/core/compatibility/sdk/8.0/runtimespecific-app-default.md

```diff
@@ -0,0 +1,74 @@
+---
+title: "Breaking change: Runtime-specific apps no longer self-contained"
+description: Learn about a breaking change in the .NET 8 SDK where apps that specify a runtime identifier are no longer self-contained by default.
+ms.date: 06/05/2023
+---
+# Runtime-specific apps no longer self-contained
+
+Runtime-specific apps, or .NET apps with a `RuntimeIdentifier`, are no longer [self-contained](../../../deploying/index.md#publish-self-contained) by default. Instead, they are [framework-dependent](../../../deploying/index.md#publish-framework-dependent) by default.
+
+This is a breaking change in the following situations:
+
+- If you deployed, distributed, or published your app and didn't explicitly add the `SelfContained` property, but also didn't require that the .NET runtime be installed on the machine for it to work. In this case, you may have relied on the previous behavior to produce a non-framework-dependent app by default.
+
+- If you rely on the IL Link tool. In this case, you'll have to take the steps described under [Recommended action](#recommended-action) to use IL Link again.
+
+Blazor WebAssembly apps relied on the previous behavior. However, the Blazor SDK now adds `SelfContained` automatically for all apps, so Blazor customers shouldn't be affected.
+
+## Previous behavior
+
+Previously, if a runtime identifier (RID) was specified (via [RuntimeIdentifier](../../../project-sdk/msbuild-props.md#runtimeidentifier)), the app was published as self-contained, even if `SelfContained` wasn't explicitly specified.
+
+In addition:
+
+- If `PublishSelfContained` wasn't explicitly set to `false`, the publish properties `PublishSingleFile` and `PublishAot` implied `PublishSelfContained` and therefore `SelfContained` during operations including `dotnet build`, `dotnet restore`, and `dotnet publish`.
+- The `PublishTrimmed` property did not imply `SelfContained`.
+- The `PublishReadyToRun` property implied `SelfContained` if `SelfContained` wasn't specified.
+
+## New behavior
+
+Starting in .NET 8, for apps that target .NET 8 or a later version, `RuntimeIdentifier` no longer implies `SelfContained` by default. Instead, apps that specify a runtime identifier will be dependent on the .NET runtime by default (framework-dependent). Apps that target .NET 7 or earlier versions aren't affected.
+
+In addition:
+
+- If `PublishSelfContained` isn't explicitly set to `false`, the publish properties `PublishTrimmed`, `PublishSingleFile`, and `PublishAot` now imply `SelfContained` during `dotnet publish` only (that is, not for `dotnet build` or `dotnet restore`).
+- The `PublishTrimmed` property also now implies `SelfContained` during `dotnet publish`.
+- The `PublishReadyToRun` property no longer implies `SelfContained` if the project targets .NET 8 or later.
+
+> [!NOTE]
+> If you publish using `msbuild t:/Publish`, the publish properties don't imply `SelfContained`.
```

Comment:
This is even weirder - and even if we need to make that for implementation reasons... do we need to document it here? I find it just plain confusing.

Reply:
I think we should document it here... I agree it's weird, but it's better that its documented and communicated than not, IMO

Reply:
Oh - I read it wrong... it's not an implementation detail as such, if I read it correctly it means that if I run `msbuild /t:Publish` it will behave differently - meaning you have to explicitly specify self-contained, right?

In that case I would word it that way - in terms of the visible functionality - talking about setting properties is difficult to grasp I think. We could have both, but I would lead with the "humanly readable" description first :-)

Also - the spelling of `/t:Publish` is wrong.